### PR TITLE
Wrapper for PHP + list update

### PR DESCRIPTION
**Pull request by hajro92**
#11

### CODE REVIEW

**Comment by hajro92** [comment]
#hacktoberfest2020

**Comment by hajro92** [comment]
Hey, for some reason my PR is marked as "Ineligible Repository", can you checkout so my PR count towards to hacktoberferst? Thank you :)

File: dist/php/businessEmailValidator.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	* @param string $email
+	* The email address we want to check
+	* @example 'john@doe.com'
+	* @return bool
+	*/
+	public function isBlocked($email)
+	{
+		filter_var($email, FILTER_VALIDATE_EMAIL) or die("Not a valid email address!");
+		$list_blocked = $this->listBlocked();
+		if(is_array($list_blocked) && filter_var($email, FILTER_VALIDATE_EMAIL))
```
Comment (musps) [inline dist/php/businessEmailValidator.php:30]: Why do you call twice `filter_var($email, FILTER_VALIDATE_EMAIL)`
any particular reason to ?

Also the return type of method `listBlocked` is array and must be array as you documented it, we don't need a double check here

File: dist/php/businessEmailValidator.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		{
+			$email_domain = explode('@', $email);
+		    $email_domain = end($email_domain);
```
Comment (musps) [inline dist/php/businessEmailValidator.php:33]: You could also improve your file indentation for better readability and maintainability

**Review by musps — CHANGES_REQUESTED** [review]
Somehelp :)

File: dist/php/businessEmailValidator.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	/**
+	* Used to load list of blocked domains from local JSON file.
+	* @access private
+	* @return array
+	*/
+	private function listBlocked()
+	{
+		$path = 'src/freeEmailService.json';
+		file_exists($path) or die("Error: file 'freeEmailService.json' not found!");
```
Comment (musps) [inline dist/php/businessEmailValidator.php:14]: Die is a bit hard here ? Maybe a Throw is a better solution to catch the error ?

File: dist/php/businessEmailValidator.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+	/**
+	* Used to check if provided email address is in list of blocked domains
+	* @access public
+	* @param string $email
+	* The email address we want to check
+	* @example 'john@doe.com'
+	* @return bool
+	*/
+	public function isBlocked($email)
```
Comment (musps) [inline dist/php/businessEmailValidator.php:26]: The return type is bool but your function calls `die x` `return x`  or `echo x`
You should only return true or false whenever the email is valid or not

**Comment by hajro92** [comment]
> Somehelp :)

Thanks for suggestion, I will try to improve my code. 😀

File: dist/php/businessEmailValidator.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+     * The email address we want to check
+     * @return bool
+     * @example 'john@doe.com'
+     */
+    public function isBlocked($email = '')
+    {
+        try {
+
+            // Domains list path
+            $path = 'src/freeEmaislService.json';
```
Comment (ankiitsinghh07) [inline dist/php/businessEmailValidator.php:18]: Hey @hajro92 ,
I checked your code and found that the mentioned filename **freeEmaislService.json** is incorrect. So, please update the same.

Comment (hajro92) [reply dist/php/businessEmailValidator.php]: Fixed

**Pull request merged**